### PR TITLE
WIP [Close #178] add no spaces rule

### LINKED ISSUE #178: Don't use spaces for indentation
> "The good equivalent of the evil `no_tabs.`" - Loïc Hoguin.

### PR DESCRIPTION
**Pull request by Euen**
[Close #178] add no spaces rule

### CODE REVIEW

File: test/examples/fail_no_spaces.erl [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+-export([one/0, two/0, three/0, four/0]).
+
+one() ->
+  not_ok. %%This lines has a spaces
+
+two() ->
+		ok. %%This one doesn't
+ 
+three() ->
+	ok. %% This one doesn't
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 12 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-module(fail_no_spaces).
+
+-export([one/0, two/0, three/0, four/0]).
+
+one() ->
+  not_ok. %%This lines has a spaces
+
+two() ->
+		ok. %%This one doesn't
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 9 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 10 has 1 trailing whitespace characters.

Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 10 has 1 trailing whitespace characters.

File: test/examples/fail_no_spaces.erl (hunk 2 above)
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 9 has a tab at column 0.

File: src/elvis_style.erl [hunk 4]
```diff
@@ -23,6 +24,8 @@
 
 -define(NO_TABS_MSG, "Line ~p has a tab at column ~p.").
 
+-define(NO_SPACES_MSG, "Line ~p has a spaces at column ~p.").
```
Comment (elbrujohalcon) [inline src/elvis_style.erl]: It should be `"Line ~p uses spaces as indentation at column ~p."`

File: src/elvis_style.erl [hunk 5]
```diff
@@ -442,6 +453,21 @@ check_no_tabs(Line, Num, _Args) ->
             {ok, Result}
     end.
 
+%% No Spaces
+
+-spec check_no_spaces(binary(), integer(), [term()]) ->
+    no_result | {ok, elvis_result:item()}.
+check_no_spaces(Line, Num, _Args) ->
+    case binary:match(Line, <<"  ">>) of
```
Comment (elbrujohalcon) [inline src/elvis_style.erl]: Spaces should be allowed if they're not at the beginning of the line.

File: test/examples/fail_no_spaces.erl [hunk 6]
```diff
@@ -0,0 +1,24 @@
+-module(fail_no_spaces).
+
+-export([one/0, two/0, three/0, four/0, five/0]).
+
+one() ->
+  not_ok. %%This lines has a spaces
+
+two() ->
+		ok. %%This one doesn't
+
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 10 has 13 trailing whitespace characters.

File: test/examples/fail_no_spaces.erl [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+four() ->
+    not_ok. %% This lines has 4 spaces
+
+five() ->
+				x:this_line_is_good(),
+				 x:this_line_is_wrong(),
+								x:this_line(is, good),
+								x:this_line(is, also,   good),
+								x:this_line(should,  nott,      x:fail( either )),
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 22 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+three() ->
+	ok. %% This one doesn't
+
+four() ->
+    not_ok. %% This lines has 4 spaces
+
+five() ->
+				x:this_line_is_good(),
+				 x:this_line_is_wrong(),
+								x:this_line(is, good),
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 20 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+             
+three() ->
+	ok. %% This one doesn't
+
+four() ->
+    not_ok. %% This lines has 4 spaces
+
+five() ->
+				x:this_line_is_good(),
+				 x:this_line_is_wrong(),
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 19 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+-export([one/0, two/0, three/0, four/0, five/0]).
+
+one() ->
+  not_ok. %%This lines has a spaces
+
+two() ->
+		ok. %%This one doesn't
+             
+three() ->
+	ok. %% This one doesn't
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 12 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+four() ->
+    not_ok. %% This lines has 4 spaces
+
+five() ->
+				x:this_line_is_good(),
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 18 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+				 x:this_line_is_wrong(),
+								x:this_line(is, good),
+								x:this_line(is, also,   good),
+								x:this_line(should,  nott,      x:fail( either )),
+        x:this_line_should(fail),
+							  x:this_line_should(fail, as, well).
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 24 has a tab at column 0.

File: test/examples/fail_no_spaces.erl [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+	ok. %% This one doesn't
+
+four() ->
+    not_ok. %% This lines has 4 spaces
+
+five() ->
+				x:this_line_is_good(),
+				 x:this_line_is_wrong(),
+								x:this_line(is, good),
+								x:this_line(is, also,   good),
```
Comment (gadgetci) [inline test/examples/fail_no_spaces.erl]: According to **Elvis**:
> Line 21 has a tab at column 0.

**Pull request merged**